### ISSUE TITLE
Selenium Grid 4 Beta - Expose Configuration option for custom response headers

### ISSUE THREAD

**Issue by XanaDublaKublaConch** [SeleniumHQ/selenium#9537]
## 🚀 Feature Proposal
Allow users to configure custom/default response headers for HTTP/S services.

## Motivation
Most webservers, including Netty's HTTP protocol server, allow users to set custom/default response headers to increase the security of web traffic. Selenium Grid Beta 4 does not appear to expose any method for setting these headers.  When deploying in secure environments or corporate environments that get scanned by vulnerability scanners, this lack of security headers will be flagged by scanning systems. Once flagged, the lack of an option to fix the issue may make Selenium Grid an undesirable testing platform for corporate security departments.

## Example
OWASP provides guidance on implementing secure headers on other HTTP servers like Apache:
https://owasp.org/www-project-secure-headers/#div-headers

Netty supports inserting headers into HTTP server responses:
https://netty.io/4.1/api/io/netty/handler/codec/http/class-use/HttpHeaders.html

**Comment by diemol** [comment]
Sounds like it could be a good idea, however we do not have time to work on this right now. Would you like to help with a PR?

**Comment by diemol** [comment]
Also, could you please provide a usage example?

**Comment by XanaDublaKublaConch** [comment]
I had a feeling this would be pretty low on the priority list.  I'm intermediate in a few programming languages, but my Java proficiency is pretty low. I played with modding Minecraft a few years ago and that was about the extent of it. I am reading Netty in Action to see if I can make some sense of Selenium/Netty code and come up with a PR, but I'm essentially learning Intermediate/Advanced Java as I read the Netty book, so it's a little ambitious. I just got the HTTP codec chapter today.

The only use case I really have is that every vulnerability scanner in existence that finds a Selenium Grid instance will flag this as it's a basic/standard security requirement for web services, so security compliance teams in Fortune 500 companies are either going to tell their devs /qa testers to remove it or ask for an exception. Exceptions generally have an expiration date, so it will have to be revisited and approved/denied on a regular basis. Without some mitigation plan, the approvals will stop.  The only other option is to front the whole installation with NGINX or some other proxy, which is a shame since Netty handles this natively and seemingly in a simple way.
